lib/class-wp-rest-widget-types-controller.php

```diff
@@ -500,6 +500,9 @@ public function encode_form_data( $request ) {
 	 * @return string
 	 */
 	private function get_widget_preview( $widget_object, $instance ) {
+		// Some third-party widgets rely on wp-admin functions so let's load them before rendering the preview.
+		require_once ABSPATH . 'wp-admin/includes/admin.php';
```

Comment:
@adamziel How about if this moves out of the private methods and put it after getting the widget's object in `encode_form_data()` like this:

```php
	public function encode_form_data( $request ) {
		$id            = $request['id'];
		$widget_object = gutenberg_get_widget_object( $id );

		if ( ! $widget_object ) {
			return new WP_Error(
				'rest_invalid_widget',
				__( 'Cannot preview a widget that does not extend WP_Widget.', 'gutenberg' ),
				array( 'status' => 400 )
			);
		}

		// Some third-party widgets rely on wp-admin functions so let's load them before invoking the widget 
		// object to avoid third-party fatal error.
		require_once ABSPATH . 'wp-admin/includes/admin.php';

		// Set the widget's number so that the id attributes in the HTML that we
		// return are predictable.
		if ( isset( $request['number'] ) && is_numeric( $request['number'] ) ) {
			$widget_object->_set( (int) $request['number'] );
		} else {
			$widget_object->_set( -1 );
		}
```

Why?

Notice the code after this point is invoking public methods on the widget. We can't be certain if any of those public methods are using wp-admin functionality. But bootstrapping the admin _before_ invoking the object's functionality, we can guard against other unknown third-party use cases.

Reply:
Good point! Let's do that – I'll update the PR.

It is also worth noting that technically the plugin could have already set up some hooks that were already executed by the time we got here, but I don't think it's possible to address every single corner here.